ndcube/utils/wcs.py

```diff
@@ -197,18 +201,32 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    dropped_coords_list = [] # Initiating new list to collect dropped coords in the process of slicing
+    # Checking item_ slices for dropped axes if any
+    for i, slice_element in enumerate(item_):
+        pix_coords = [0] * len(item_) # Setting up a list of pixel coords as input to all_pix2world
+        if missing_axis[i] is False and (slice_element.stop - slice_element.start) == 1:
+            pix_coords[i] = slice_element.start # Enter pixel coordinate for this axis. 
+                # Since we are only dealing with independent axes the other axes can remain at 0.
+            real_world_coords = wcs.all_pix2world(*pix_coords, 0)[i]
+                # Unravel the arguments in the pix_coords array using the * prefix.
+                # Added in index to obtain the i-th element in the resultant real world coords list of arrays. 
+            ctype = wcs.wcs.ctype[i] # Added an index to get the ctype name for the i-th element of wcs's ctype list.   
+            dropped_coords_list.append((ctype, None, real_world_coords))
+    dropped_coords = tuple(dropped_coords_list) 
```

Comment:
I don't think there's any need to convert this to a ```tuple``` of ```tuple```s.  A ```list``` of ```tuples``` is fine.

Reply:
Yup, was wondering about this too (but had been hesitating to mention it before)... I actually wanted to use a tuple of tuples to make sure the object is immutable. In hindsight, I think your suggesting using a list of tuples might render it easier to manipulate the list.